cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectorAssemblyOperator.java

```diff
@@ -0,0 +1,103 @@
+import io.strimzi.api.kafka.model.DoneableKafkaConnector;
+import io.strimzi.api.kafka.model.KafkaConnector;
+import io.strimzi.api.kafka.model.KafkaConnectorSpec;
+import io.strimzi.operator.common.AbstractOperator;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.model.NamespaceAndName;
+import io.strimzi.operator.common.operator.resource.CrdOperator;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Operator for a {@code KafkaConnector} CRs.
+ * This is managed via the {@code KafkaConnectAssemblyOperator}, rather than directly by the {@code ClusterOperator}.
+ */
+public class KafkaConnectorAssemblyOperator extends
+        AbstractOperator<KafkaConnector, CrdOperator<KubernetesClient, KafkaConnector, KafkaConnectorList, DoneableKafkaConnector>> {
+    private static final Logger log = LogManager.getLogger(KafkaConnectorAssemblyOperator.class.getName());
+    private final LabelSelector selector;
+    private KafkaConnectApi apiClient;
+
+    /**
+     * @param vertx       The Vertx instance
+     * @param selector    The selector for finding connectors
+     * @param crdOperator The CRD operator
+     * @param apiClient   The Kafka Connect API client
+     */
+    public KafkaConnectorAssemblyOperator(Vertx vertx,
+                                          LabelSelector selector,
+                                          CrdOperator<KubernetesClient, KafkaConnector, KafkaConnectorList, DoneableKafkaConnector> crdOperator,
+                                          KafkaConnectApi apiClient) {
+        super(vertx, KafkaConnector.RESOURCE_KIND, crdOperator);
+        Objects.requireNonNull(selector);
+        this.selector = selector;
+        this.apiClient = apiClient;
+    }
+
+    @Override
+    public Optional<LabelSelector> selector() {
+        return Optional.of(selector);
+    }
+
+    @Override
+    protected Future<Void> createOrUpdate(Reconciliation reconciliation, KafkaConnector assemblyResource) {
+        try {
+            String namespace = reconciliation.namespace();
+            String name = reconciliation.name();
+
+            log.debug("{}: Creating/Updating Kafka Connector", reconciliation, name, namespace);
+
+            KafkaConnectorSpec spec = assemblyResource.getSpec();
+            JsonObject connectorConfigJson = new JsonObject()
+                    .put("connector.class", spec.getClassName())
+                    .put("tasks.max", spec.getTasksMax());
+            spec.getConfig().forEach(cf -> connectorConfigJson.put(cf.getName(), cf.getValue()));
+
+            return apiClient.createOrUpdatePutRequest(name, connectorConfigJson)
```

Comment:
Would it make sense to do a `validate` call upfront? I guess that not, but wanted to raise the idea.

Reply:
I don't see any compelling advantages. I guess we could better process validation failures to give a _slightly_ better error message in the `status`, but I'm not sure that it's worth it at the moment.

Reply:
Ok, that's what I thought. Thanks!